### PR TITLE
Feature/raven export wrapper

### PR DESCRIPTION
**Pull request by shaing10**
## Summary

- Adds `scripts/raven_export.py` — exports soundbay checkpoints (ResNet182D, Squeezenet2D) to Raven Intelligence `.ravenmodel` packages with preprocessing baked in
- Supports TorchScript (recommended) and ONNX export formats
- Re-implements PeakNormalize, UnitNormalize, and LibrosaPcen as `nn.Module`s for TorchScript compatibility
- Adds `--compensate-for-sigmoid` flag (recommended): Raven applies sigmoid to model outputs internally, so this option outputs `inverse_sigmoid(softmax(logits))` — Raven's sigmoid then recovers exact softmax probabilities that sum to 1.0

## Usage

**Recommended** (until Raven publishes a new version with more DOFs in their post-processing logic): compensate for Raven's sigmoid post-processing

```bash
python scripts/raven_export.py /path/to/checkpoint/best.pth \
    --output-dir ~/Raven\ Workbench/Raven\ Intelligence/Models/ \
    --name My_Detector \
    --compensate-for-sigmoid
```

**Raw logits** (no activation, Raven applies sigmoid):

```bash
python scripts/raven_export.py /path/to/checkpoint/best.pth \
    --name My_Detector --no-softmax
```

## Why some modules are re-implemented

TorchScript export requires all operations to be `nn.Module` instances or pure torch ops. Several soundbay preprocessors can't be traced as-is:

| Module | Why re-implemented | Approach |
|---|---|---|
| `PeakNormalize` | Plain class with `__call__`, not `nn.Module` — TorchScript can't trace it | Thin `nn.Module` wrapper with identical logic (verified: 0.0 diff) |
| `UnitNormalize` | Same — not `nn.Module` | Same approach (verified: 0.0 diff) |
| `LibrosaPcen` | Calls `librosa.core.pcen` (NumPy/SciPy) — can't run in TorchScript | Full reimplementation in pure PyTorch with `torch.jit.script` for the IIR loop. Matches librosa's filter coefficients and `lfilter_zi` initialization exactly |
| `MinFreqFiltering` | Uses `numpy.floor` | Replaced with `nn.Identity` (validated: only `min_freq=0` configs are supported; errors on non-zero) |

## Test plan

- [x] Exported barks detector with all 3 modes (softmax, no-softmax, compensate-for-sigmoid)
- [x] Verified compensated model scores sum to exactly 1.0 after Raven's sigmoid
- [x] Confirmed detection agreement across all export modes on real WAV file
- [x] Tested in Raven Intelligence with TH=0.2

### CODE REVIEW

**Review by mosheman5 — COMMENTED** [review]
looks good
do we still new --compensate-for-sigmoid in the default/recommended command?

**Comment by shaing10** [comment]
> looks good do we still new --compensate-for-sigmoid in the default/recommended command?

image
seems like the new release solved it, I'll change the default

**Review by mosheman5 — APPROVED** [review]
